From b9b6137a6b7553f3bd1a394a8406ca501b343c97 Mon Sep 17 00:00:00 2001
From: Paradoxon <two4god@gmail.com>
Date: Sat, 9 Jul 2016 16:07:26 +0200
Subject: [PATCH] WebPositive - dont load history if a ref is recived

---
 src/apps/webpositive/BrowserApp.cpp | 46 ++++++++++++++++++-------------------
 1 file changed, 22 insertions(+), 24 deletions(-)

diff --git a/src/apps/webpositive/BrowserApp.cpp b/src/apps/webpositive/BrowserApp.cpp
index ef865af..329556f 100644
--- a/src/apps/webpositive/BrowserApp.cpp
+++ b/src/apps/webpositive/BrowserApp.cpp
@@ -230,34 +230,32 @@ BrowserApp::ReadyToRun()
 		delete fLaunchRefsMessage;
 		fLaunchRefsMessage = NULL;
 	}
-
-	BMessage archivedWindow;
-	for (int i = 0; fSession->FindMessage("window", i, &archivedWindow) == B_OK;
-		i++) {
-
-		BRect frame = archivedWindow.FindRect("window frame");
-		BString url;
-		archivedWindow.FindString("tab", 0, &url);
-		BrowserWindow* window = new(std::nothrow) BrowserWindow(frame,
-			fSettings, url, fContext);
-
-		if (window != NULL) {
-			window->Show();
-			pagesCreated++;
-
-			for (int j = 1; archivedWindow.FindString("tab", j, &url) == B_OK;
-				j++) {
-				printf("Create %d:%d\n", i, j);
-				_CreateNewTab(window, url, false);
+	if (pagesCreated == 0){
+		BMessage archivedWindow;
+		for (int i = 0; fSession->FindMessage("window", i, &archivedWindow) == B_OK;
+			i++) {
+			BRect frame = archivedWindow.FindRect("window frame");
+			BString url;
+			archivedWindow.FindString("tab", 0, &url);
+			BrowserWindow* window = new(std::nothrow) BrowserWindow(frame,
+				fSettings, url, fContext);	
+
+			if (window != NULL) {
+				window->Show();
 				pagesCreated++;
+
+				for (int j = 1; archivedWindow.FindString("tab", j, &url) == B_OK;
+					j++) {
+					printf("Create %d:%d\n", i, j);
+					_CreateNewTab(window, url, false);
+					pagesCreated++;
+				}
 			}
 		}
+		if (pagesCreated == 0)	
+			_CreateNewWindow("", fullscreen);
+		PostMessage(PRELOAD_BROWSING_HISTORY);
 	}
-
-	if (pagesCreated == 0)
-		_CreateNewWindow("", fullscreen);
-
-	PostMessage(PRELOAD_BROWSING_HISTORY);
 }
 
 
-- 
2.7.0

